process/headerCheck/headerSignatureVerify.go

```diff
@@ -171,7 +178,16 @@ func (hsv *HeaderSigVerifier) VerifySignature(header data.HeaderHandler) error {
 		return err
 	}
 
-	return hsv.VerifySignatureForHash(header, hash, header.GetPubKeysBitmap(), header.GetSignature())
+	bitmap := header.GetPubKeysBitmap()
+	sig := header.GetSignature()
+	if hsv.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, headerCopy.GetEpoch()) {
```

Comment:
i think that header signature verifier component should not know if we want to handle current or previous header info; can we use only `VerifySingatureForHash` and decide from the caller which header info to use?

Reply:
how about other components where `VerifySignature` is being used? does this also fit with the check from process interceptors?

Reply:
yes, it should be compatible in all places. We can keep it as it is for compatibility. After this release, a refactor will come in the next one probably and completely remove EquivalentMessagesFlag and only keep the new version.